server.js

```diff
@@ -1,24 +1,67 @@
-const express = require('express');
+const express = require("express");
 const app = express();
 
+// Port
+let PORT = 5001;
+app.listen(PORT, () => {
+  console.log(`Example app listening on port ${PORT}!`);
+});
+
+// Middleware to parse
+app.use(express.json());
+
 let fruits = [
-  { id: "1", name: 'Banana', color: 'yellow'},
-  { id: "2", name: 'Cherry', color: 'red'},
-  { id: "3", name: 'Lemon', color: 'yellow'},
-  { id: "4", name: 'Apple', color: 'green'},
-  { id: "5", name: 'Orange', color: 'orange'}
-]
-
-app.get('/fruits', (req, res) => {
-  res.json(fruits)
+  { id: "1", name: "Banana", color: "yellow" },
+  { id: "2", name: "Cherry", color: "red" },
+  { id: "3", name: "Lemon", color: "yellow" },
+  { id: "4", name: "Apple", color: "green" },
+  { id: "5", name: "Orange", color: "orange" },
+];
+
+// get  all fruits
+app.get("/fruits", (req, res) => {
+  res.json(fruits);
 });
 
-app.get('/fruits/:id', (req, res) => {
-  res.json({})
-})
+// more specific routes go on the top
+app.get("/fruits/plain", (req, res) => {
+  console.log("ROUTE CALLED!!");
+  const fruitsArray = fruits.map((fruit) => fruit.name);
+  res.json(fruitsArray);
+});
+
+// get one fruit by id
+app.get("/fruits/:id", (req, res) => {
+  const { id } = req.params;
+  console.log(`ID sent to us: ${id}`);
+  let fruitFound = fruits.find((fruit) => fruit.id == id);
+  res.json(fruitFound);
+});
+
+// POST adding a fruit DO I NEED REQ.BODY?
+app.post("/fruits", (req, res) => {
+  const newFruit = req.body;
+  fruits.push(newFruit);
```

Comment:
Here it would be cool, if we also create an ID for each new fruit, before we push it on our array.

So something like this:
newFruit.id = Date.now().toString()
fruits.push( newFruit )
res.json( newFruit )

Because then we can also change these new items again later, e.g. by PATCH or DELETE where we always need to specify the id.